### ISSUE TITLE
XCode IB does not see the MACachedImageView class

### ISSUE THREAD

**Issue by JanC** [swissmanu/MACachedImageView#1]
Hi,
I installed MACachedImageView using CocoaPods. I can instantiate a MACachedImageView in code fine but when I want to say to the Interface Builder that the class of a UIImageView is a MACachedImageView, it does not work

![Screen Shot 2013-02-19 at 7 34 45 PM](https://f.cloud.github.com/assets/707703/172661/4cb03902-7ac3-11e2-802f-d4666c213b03.png)

what did I do wrong?

cheers

**Comment by swissmanu** [comment]
hi jan

i have the same problem from time to time when using custom classes in
interface builder.

have you tried to enter the class name manually? usually it will work
anyway,
even if autocompletion fails.

let me know if this helps.

cheers,
manu

Am 19.02.2013 um 19:36 schrieb Jan Chaloupecky notifications@github.com:

Hi,
I installed MACachedImageView using CocoaPods. I can instantiate a
MACachedImageView in code fine but when I want to say to the Interface
Builder that the class of a UIImageView is a MACachedImageView, it does not
work

[image: Screen Shot 2013-02-19 at 7 34 45
PM]https://f.cloud.github.com/assets/707703/172661/4cb03902-7ac3-11e2-802f-d4666c213b03.png

what did I do wrong?

cheers

—
Reply to this email directly or view it on
GitHubhttps://github.com/swissmanu/MACachedImageView/issues/1.

**Comment by JanC** [comment]
Hi,
thanks for the fast answer.

If I enter it manually, hit enter then select another UI element then come back to my image, it is set back to UIImageView.

It works only If I set the custom class in the story board xml file 

```
<imageView userInteractionEnabled="NO" tag="102" contentMode="scaleToFill" image="awards_m_r01.png" id="jYr-Aw-l85" customClass="MACachedImageView">
```

**Comment by swissmanu** [comment]
i see... try to add plain UIView to your storyboard and set the class of
that one.

MACachedImageView inherits from the UIView main class instead of the
UIImageView subclass.

have to mention that in the readme probably...

cheers,
manu

**Comment by JanC** [comment]
ok thanks.

That's not very handy though cos now I have to replace all my UIImageViews with UIViews in the IB :(

**Comment by swissmanu** [comment]
using the UIImageView as subclassing entry point would definitely make sense.
i'll have a look into it. maybe a next release will fix this semantic issue.

sorry for the inconvenience.

**Comment by JanC** [comment]
that would be nice ;) Thanks

**Comment by swissmanu** [comment]
the just released version `1.1.0` of `MACachedImageView` fixes this shortcoming. `MACachedImageView` is a subclass of `UIImageView` now.

thanks for your feedback @JanC :)
